lib/elixir/pages/anti-patterns/process-anti-patterns.md

```diff
@@ -290,3 +290,42 @@ iex> Supervisor.restart_child(App.Supervisor, Counter)
 iex> Counter.get(Counter) # After the restart, this process can be used again
 0
 ```
+
+## Sending unnecessary data
+
+#### Problem
+
+Sending a message to a process can be an expensive operation if it is big enough, as messages will be fully copied to the receiving process, which is both CPU and memory intensive. This is due to Erlang's "share nothing" architecture where each process has its own memory, simplifying and speeding up garbage collection.
+This is more obvious when using `send/2`, `GenServer.call/3`, or the initial data in `GenServer.start_link/3`. Notably this also happens when using `spawn/1`, `Task.async/1`, `Task.async_stream/3`, etc. It is more subtle here as the anonymous function passed to these functions captures the variables it references in its closure; this means that all data will be copied over. By doing this, you can accidentally send way more data to a process than you actually need.
+
+#### Example
+
+To depict the problem, let's imagine you were to implement some simple reporting of IP addresses that made requests against your application. You want to do this asynchronously and not block processing, so you decide to use `spawn/1`. It may seem like a good idea to hand over the whole connection because we might need more data later. However passing the connection results in copying a lot of unnecessary data like the request body, params, etc.
+
+```elixir
+# log_request_ip send the ip to some external service
+spawn(fn -> log_request_ip(conn) end)
+```
+
+Please note that this problem also occurs if you think you may just be accessing the relevant parts:
+
+```elixir
+spawn(fn -> log_request_ip(conn.remote_ip) end)
+```
+
+This will still copy over all of `conn`.
```

Comment:
More context. Thoughts?

```suggestion
This will still copy over all of `conn`, because the `conn` variable is being captured as a closure inside the spawned function. That function then extracts the `remote_ip` field, but only after the whole `conn` has been copied over.
```

Reply:
Is "as a closure" right here? I thought it was "within the closure of the spawned function" - as in, the function has a closure and storing the `conn` is part of it - the `conn` is not the closure itself :thinking: 

Adding now, we can rework later if needed - already provides more value, thank you!

Reply:
(unresolved to clear the point about closures)

Reply:
Your version is more technical but I wanted to try and keep it simple 😉